lib/manager/components/deployment/CustomConfig.js

```diff
@@ -38,24 +44,49 @@ export default class CustomConfig extends Component<{
 }, {[string]: any}> {
   state = {}
 
-  _toggleCustomConfig = (evt: SyntheticInputEvent<HTMLInputElement>) => {
-    const {deployment, updateDeployment} = this.props
-    const {name} = evt.target
-    const value = deployment[name]
-      ? null
-      : name === 'customBuildConfig'
-        ? SAMPLE_BUILD_CONFIG
-        : SAMPLE_ROUTER_CONFIG
-    updateDeployment(deployment, {[name]: value})
+  getName = (option?: string) => {
+    let {name} = this.props
+    if (option === CONFIG_OPTIONS.url) {
+      name += 'Url'
+    }
+    return name
   }
 
-  _onChangeConfig = (evt: SyntheticInputEvent<HTMLInputElement>) =>
-    this.setState({[this.props.name]: evt.target.value})
-
-  _onSaveConfig = () => {
+  _toggleCustomConfig = (evt: SyntheticInputEvent<HTMLInputElement>, option?: string) => {
     const {deployment, name, updateDeployment} = this.props
+    let value = 'https://'
+    if (name === 'customBuildConfig' && option === CONFIG_OPTIONS.custom) value = deployment[name] || SAMPLE_BUILD_CONFIG
+    if (name === 'customRouterConfig' && option === CONFIG_OPTIONS.custom) value = deployment[name] || SAMPLE_ROUTER_CONFIG
+
+    // If no option, clear everything
+    if (!option) {
+      updateDeployment(deployment, {[name + 'Url']: null, [name]: null})
+    }
+
+    // If custom content, clear URL
+    if (option === CONFIG_OPTIONS.custom) {
+      updateDeployment(deployment, {[name + 'Url']: null, [name]: value})
+    }
+
+    // If custom URL, clear content
+    if (option === CONFIG_OPTIONS.url) {
+      updateDeployment(deployment, {[name + 'Url']: value})
```

Comment:
Should we also clear `[name]: null ` here?

Reply:
No! This allows the custom content to be preserved if we switch back